components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1032,469 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
+    // TODO: log-surgeon needs to be refactored to allow direct usage of DFA/NFA.
+    // TODO: DFA creation isn't optimized at all.
+    auto const search_string_dfa = ByteLexer::nfa_to_dfa(nfa);
+    auto const& schema_dfa = lexer.get_dfa();
+
+    // Get variable types in the intersection of substring and compression DFAs.
+    return {schema_dfa->get_intersect(search_string_dfa), contains_wildcard};
+}
+
+vector<string>
+Grep::generate_logtype_strings(vector<QueryLogtype>& query_logtypes, ByteLexer& lexer) {
+    vector<string> logtype_strings;
+    logtype_strings.reserve(query_logtypes.size());
+    for (QueryLogtype const& query_logtype : query_logtypes) {
+        // Convert each query logtype into a set of logtype strings. Logtype strings are used in the
+        // sub query as they have the correct format for comparing against the archive. Also, a
+        // single query logtype might represent multiple logtype strings. While static text converts
+        // one-to-one, wildcard variables that may be encoded have different logtype strings when
+        // comparing against the dictionary than they do when comparing against the segment.
+        auto& logtype_string = logtype_strings.emplace_back();
+        for (uint32_t i = 0; i < query_logtype.get_logtype_size(); i++) {
+            auto const logtype_value = query_logtype.get_logtype_value(i);
+            auto const& raw_string = query_logtype.get_query_string(i);
+            auto const is_encoded_with_wildcard = query_logtype.get_is_encoded_with_wildcard(i);
+            auto const var_has_wildcard = query_logtype.get_var_has_wildcard(i);
+            if (std::holds_alternative<char>(logtype_value)) {
+                logtype_string.push_back(std::get<char>(logtype_value));
+            } else {
+                auto& schema_type = lexer.m_id_symbol[std::get<int>(logtype_value)];
+                encoded_variable_t encoded_var;
+
+                // If this logtype contains wildcard variables that are being compared against the
+                // dictionary, create a duplicate logtype that will compare against segment if the
+                // variable may be encoded there instead.
+                if (false == is_encoded_with_wildcard && var_has_wildcard
+                    && ("int" == schema_type || "float" == schema_type))
+                {
+                    auto new_query_logtype = query_logtype;
+                    new_query_logtype.set_is_encoded_with_wildcard(i, true);
+                    query_logtypes.push_back(new_query_logtype);
+                }
+                if (is_encoded_with_wildcard) {
+                    if ("int" == schema_type) {
+                        LogTypeDictionaryEntry::add_int_var(logtype_string);
+                    } else if ("float" == schema_type) {
+                        LogTypeDictionaryEntry::add_float_var(logtype_string);
+                    }
+                } else if (false == var_has_wildcard && "int" == schema_type
```

Comment:
It's possible that a wildcard-containing token matches both the `int` and `float` regexes, right? In which case, we should actually generate a `QueryLogtype` for each one.

Reply:
It is possible for something like `"*10000"` to match both `int` and `float`, but this is already captured in query_logtypes having a logtype for both cases. So, a logtype string should be generated for each logtype respectively. `"Wildcard variable"` unit test demonstrates this case.